src/body-parts/BodyPart.cpp

```diff
@@ -1,330 +1,441 @@
 
 /**
- * @return The offset of the body part
+ * @return The depth of the body part
  */
-[[nodiscard]] Vector4 BodyPart::getOffset() const
+[[nodiscard]] float BodyPart::getDepth() const
 {
-    return _offset;
+    return _depth;
 }
 
 /**
- * @return The position of the body part
+ * @return The height of the body part
  */
-[[nodiscard]] Vector4 BodyPart::getPosition() const
+[[nodiscard]] float BodyPart::getHeight() const
 {
-    return _position;
+    return _height;
 }
 
 /**
- * @return The angle on the X axis of the body part
+ * @return The width of the body part
  */
-[[nodiscard]] double BodyPart::getAngleX() const
+[[nodiscard]] float BodyPart::getWidth() const
 {
-    return _angleX;
+    return _width;
 }
 
 /**
- * @return The angle on the Y axis of the body part
+ * @return The matrix stack of the body part
  */
-[[nodiscard]] double BodyPart::getAngleY() const
+[[nodiscard]] Matrix4 BodyPart::getMatrixStack() const
 {
-    return _angleY;
+    return _matrixStack.top();
 }
 
 /**
- * @return The angle on the Z axis of the body part
+ * @return The transformation matrix of the body part
  */
-[[nodiscard]] double BodyPart::getAngleZ() const
+[[nodiscard]] Matrix4 BodyPart::getTransformationMatrix() const
 {
-    return _angleZ;
+    // Translation to bring the object to the pivot point
+    const Matrix4 translationToPivot = Matrix4::createTranslationMatrix(-_pivotPoint.getX(),
+                                                                        -_pivotPoint.getY(),
+                                                                        -_pivotPoint.getZ());
+
+    // Apply all transformation : translation, rotation, etc.
+    const Matrix4 combinedTransformation = _translationMatrix * _rotationMatrix;
+
+    // Translation to bring back the object to its origin point
+    const Matrix4 translationBackFromPivot = Matrix4::createTranslationMatrix(
+        _pivotPoint.getX(),
+        _pivotPoint.getY(),
+        _pivotPoint.getZ());
+
+    return translationBackFromPivot * combinedTransformation * translationToPivot;
 }
 
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// Setters
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
 /**
- * @return The parent of the body part
+ * Set the color of the body part using RGB values.
+ *
+ * @param red   The new red value
+ * @param green The new green value
+ * @param blue  The new blue value
+ *
+ * @return itself
  */
-[[nodiscard]] BodyPart* BodyPart::getParent() const
+BodyPart& BodyPart::setColor(const short int red, const short int green, const short int blue)
 {
-    return _parent;
+    _red = red;
+    _green = green;
+    _blue = blue;
+    return *this;
 }
 
 /**
- * @return The children of the body part
+ * Set the depth of the body part.
+ *
+ * @param depth The new red value
```

Comment:
```suggestion
 * @param depth The new depth value
```
Les fameux ctrl+c ctrl+v 👀

Reply:
Je vois pas de quoi tu parles